plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/librarylinker/LibraryLinker.java

```diff
@@ -238,44 +237,55 @@ public boolean removeOldLibVersion(final IFolder libDirectory) {
 		return isNewVersion;
 	}
 
-	@Override
-	public void createTypeEntriesManually(final IContainer container) {
-		try {
-			final IResource[] members = container.members();
-			for (final IResource resource : members) {
-				if (resource instanceof final IFolder folder) {
-					createTypeEntriesManually(folder);
-				}
-				if (resource instanceof final IFile file) {
-					typeLibrary.createTypeEntry(file);
-				}
+	private void createTypeEntriesManually(final IContainer container) {
+		final IResourceVisitor visitor = resource -> (switch (resource) {
+		case final IFolder folder -> true;
+		case final IFile file -> {
+			final TypeEntry entry = TypeEntryFactory.INSTANCE.createTypeEntry(file);
+			final TypeEntry oldEntry = cachedTypes.get(entry.getFullTypeName());
+			if (oldEntry != null) {
+				FordiacResourceChangeListener.updateTypeEntry(file, oldEntry);
+				cachedTypes.remove(entry.getFullTypeName());
+			} else {
+				typeLibrary.createTypeEntry(file);
 			}
+			yield false;
+		}
+		default -> false;
+		});
+		try {
+			container.accept(visitor);
 		} catch (final CoreException e) {
 			FordiacLogHelper.logError(e.getMessage(), e);
 		}
-
 	}
 
-	@Override
-	public void updateFBInstancesWithNewTypeVersion() {
+	private void cleanupOldLibraryVersion() {
```

Comment:
in this method you are doing two things:

1. cleaning up the deleted types
2. informing the user

I think the second should be moved to its own method. Because here I would never search for it.

Reply:
split it into two methods

Reply:
thx